lib/lita/handlers/locker_resources.rb

```diff
@@ -46,22 +47,49 @@ class LockerResources < Handler
       )
 
       def list(response)
-        after 0 do
-          should_rate_limit = false
-
-          Resource.list.each_slice(5) do |slice|
-            if should_rate_limit
-              sleep 3
-            else
-              should_rate_limit = true
-            end
-
-            slice.each do |r|
-              res = Resource.new(r)
-              response.reply(t('resource.desc', name: r, state: res.state.value))
-            end
+        list = Resource.list
+        count = list.count
+
+        begin
+          page = Integer(response.extensions[:kwargs][:page].to_s, 10)
+        rescue ArgumentError
+          response.reply t("list.invalid_page_type")
+
+          return
+        end
+
+        pages = (count / config.per_page).ceil + 1
```

Comment:
This line looks fishy to me. I would feel more confident if I saw tests for edge-cases, like where count was evenly divisible by `per_page`, or where count = 0, count = `per_page`... that kinda thing